src/server/frontend_xwayland/xwayland_surface.cpp

```diff
@@ -183,6 +228,19 @@ auto property_handler(
 {
     return property_handler<T>(connection, window, property, mf::XCBConnection::Handler<T>{std::move(handler)});
 }
+
+template<typename T>
+auto optional_from(bool return_value, T&& value) -> std::optional<T>
+{
+    if (return_value)
+    {
+        return std::optional<T>(value);
```

Comment:
Is there any point in taking an rvalue reference if we don't `std::move()` from it?

Reply:
Fixed